os/fs/vfs/fs_pread.c

```diff
@@ -112,7 +112,10 @@ ssize_t file_pread(FAR struct file *filep, FAR void *buf, size_t nbytes, off_t o
 	/* Then perform the read operation */
 
 	ret = file_read(filep, buf, nbytes);
-	errcode = get_errno();
+	if (ret < 0) {
```

Comment:
I think set_errno is already being called in file_read & file_seek. how about below?
```
if (ret < 0) {
    ret = ERROR;
}
...
//set_errno(errcode);  //remove this line!